### DISCUSSION TITLE
offline map integration difficulties

### DISCUSSION THREAD

**Discussion by scorpiomac** [fleaflet/flutter_map#1653] [Q&A]
For 03 weeks I've been stuck loading the card.
I've tried Mbtile files without success, PNG images too, please can someone help me. It's very urgent

**Comment by JaffaKetchup** [comment]
Hey @scorpiomac,
You'll need to be a little more detailed about what issues you've faced and what you've already tried.

**Reply by scorpiomac** [reply]
My main.dart code :

import 'package:flutter/material.dart';
import 'package:flutter_map/flutter_map.dart';
import 'package:flutter_map_example/widgets/drawer.dart';
import 'package:latlong2/latlong.dart';

class OfflineMapPage extends StatelessWidget {
static const String route = '/offline_map';

const OfflineMapPage({Key? key}) : super(key: key);

@OverRide
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(title: const Text('Offline Map')),
drawer: buildDrawer(context, route),
body: Padding(
padding: const EdgeInsets.all(8),
child: Column(
children: [
const Padding(
padding: EdgeInsets.only(top: 8, bottom: 8),
child: Text(
'This is an offline map that is showing Anholt Island, Denmark.'),
),
Flexible(
child: FlutterMap(
options: MapOptions(
initialCenter: const LatLng(56.704173, 11.543808),
minZoom: 12,
maxZoom: 14,
cameraConstraint: CameraConstraint.containCenter(
bounds: LatLngBounds(
const LatLng(56.7378, 11.6644),
const LatLng(56.6877, 11.5089),
),
),
),
children: [
TileLayer(

                tileProvider: AssetTileProvider(),
                maxZoom: 14,
                urlTemplate: 'assets/map/tiles{z}/{x}/{y}.png',
              ),
            ],
          ),
        ),
      ],
    ),
  ),
);
}
}


I have this issues :
error : flutter: Unable to load asset: "assets/map/anholt_osmbright/12/2046/1361.png".
flutter: Exception: Asset not found
flutter: Exception: Asset not found
flutter: Unable to load asset: "assets/map/tiles/12/2047/1360.png".
flutter: Exception: Asset not found
flutter: Unable to load asset: "assets/map/tiles/12/2045/1363.png".
flutter: Exception: Asset not found
flutter: Unable to load asset: "assets/map/tiles/12/2047/1363.png".
flutter: Exception: Asset not found

**Reply by JaffaKetchup** [reply]
Whats in your pubspec assets section?

**Reply by scorpiomac** [reply]
assets:
    # - assets/map/anholt_osmbright/12/2177/
    # - assets/map/anholt_osmbright/12/2178/
    # - assets/map/anholt_osmbright/12/2179/
    # - assets/map/anholt_osmbright/12/2180/
    # - assets/map/anholt_osmbright/12/2181/
    # - assets/map/anholt_osmbright/13/4354/
    # - assets/map/anholt_osmbright/13/4355/
    # - assets/map/anholt_osmbright/13/4356/
    # - assets/map/anholt_osmbright/13/4357/
    # - assets/map/anholt_osmbright/13/4358/
    # - assets/map/anholt_osmbright/13/4359/
    # - assets/map/anholt_osmbright/13/4360/
    # - assets/map/anholt_osmbright/13/4361/
    # - assets/map/anholt_osmbright/13/4362/
    # - assets/map/anholt_osmbright/13/4363/
    # - assets/map/anholt_osmbright/14/8709/
    # - assets/map/anholt_osmbright/14/8710/
    # - assets/map/anholt_osmbright/14/8711/
    # - assets/map/anholt_osmbright/14/8712/
    # - assets/map/anholt_osmbright/14/8713/
    # - assets/map/anholt_osmbright/14/8714/
    # - assets/map/anholt_osmbright/14/8715/
    # - assets/map/anholt_osmbright/14/8716/
    # - assets/map/anholt_osmbright/14/8717/
    # - assets/map/anholt_osmbright/14/8718/
    # - assets/map/anholt_osmbright/14/8719/
    # - assets/map/anholt_osmbright/14/8720/
    # - assets/map/anholt_osmbright/14/8721/
    # - assets/map/anholt_osmbright/14/8722/
    # - assets/map/anholt_osmbright/14/8723/
    # - assets/map/anholt_osmbright/14/8724/
    # - assets/map/anholt_osmbright/14/8725/
    # - assets/map/anholt_osmbright/14/8726/
    - assets/map/anholt_osmbright/
    - assets/map/epsg3413/amsr2.png
    - assets/tiles/
    - assets/

**Comment by ibrierley** [comment]
That looks a bit weird, as though it's trying to use a part of the map that it shouldn't have access to (and doesn't exist). I'm not really clear why that would be, as the code looks kinda ok (assuming you have moved the assets folder  from assets/map/anholt_osmbright maybe to assets/map/tiles but I think thats a redherring, the tile x,y coordinates look incorrect, but I may be wrong).

What type of device and setup are you using ? (not sure why this should matter, it just feels something is odd).

**Reply by scorpiomac** [reply]
You're right.
I use an iPhone 14 plus and 14 pro max but always the same error

**Comment by ibrierley** [comment]
Does it do this on a web version ? Just wondering if it's possible it's some retina mode thing which loads in more tiles which aren't available ? What happens if you play with retinaMode in the tile layer ? (again, I don't think it is, but just exploring some thoughts)

**Reply by scorpiomac** [reply]
I have the same problem on the web.
Concerning the retinaMode modification.
Can you please tell me how to do it?

**Reply by scorpiomac** [reply]
If you know of another way to integrate an offline map into flutter with a code that works, I'm also interested.

**Comment by scorpiomac** [comment]
If you know of another way to integrate an offline map into flutter with a code that works, I'm also interested.

**Comment by ibrierley** [comment]
You'd just set retinaMode=true/false I think in the tile layer options. I don't think this is the issue, as I think you'd have to change the tileUrl, but kinda clutching at straws. I'm just a bit confused why its trying to load 12/2046/1361.png as an example, when mine doesn't.

**Reply by scorpiomac** [reply]
I'm just as confused as you are.
When I downloaded the example git repo everything worked.
So I added a new tiles folder in assets where I added my own map that I downloaded.
When I ran it, it started giving me an error. I was already lost.
Then I changed the path back to assets/map/anholt_osmbright but that didn't change anything.
I'm still having the same problem with files that were working just 10 minutes ago.

**Comment by ibrierley** [comment]
In that case, start with a fresh checkout again, change one thing at a time (then change it back) until its isolated. It's not clear what the map is that you downloaded or what that is, so it could be worth posting more about that if that is indeed the problem.
